pkg/report/report.go

```diff
@@ -287,26 +261,29 @@ func (reporter *Reporter) ReportToGuiltyFile(title string, report []byte) string
 	return ii.extractGuiltyFileRaw(title, report)
 }
 
-func extractReportType(rep *Report) Type {
+func (reporter *Reporter) extractReportType(rep *Report) crash.Type {
+	if rep.Type != crash.UnknownType {
```

Comment:
The current logic to deal with BUG is quite complex and hard to follow.

I think we can remove all special cases and custom matching code if we will be able to distinguish Unknown from Unspecified type.
I.e. we either add Unspecified (which will be an empty string) and then Bug oops says crash.Bug, but some concrete oopsFormat's will be able to override it to crash.Unknown. Or we add something like UnknownOverride, which we can add to oopsFormat to override oops type to Unknown.